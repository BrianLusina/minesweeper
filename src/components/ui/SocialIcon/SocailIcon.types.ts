export type SocialIconProps = {
  link: string;
  icon: string;
  name: string;
};
