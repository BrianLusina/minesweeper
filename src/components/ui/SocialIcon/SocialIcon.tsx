import { FunctionComponent } from 'react';
import { SocialIconProps } from './SocailIcon.types';

const SocialIcon: FunctionComponent<SocialIconProps> = ({ name, link, icon }) => {
  return (
    <a
      href={link}
      className={`icon brands fa-${icon.toLowerCase()}`}
      target="_blank"
      rel="noopener noreferrer"
    >
      <span className="label">{name}</span>
    </a>
  );
};

export default SocialIcon;
