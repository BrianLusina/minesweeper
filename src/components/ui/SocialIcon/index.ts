import Socialicon from './SocialIcon';

export default Socialicon;
