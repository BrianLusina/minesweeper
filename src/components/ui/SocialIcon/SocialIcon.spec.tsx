import { render, screen } from '@testing-library/react';
import SocialIcon from './SocialIcon';

describe('SocialIcon', () => {
  it('should render', () => {
    const name = 'Twitter';
    const link = 'https://twitter.com/usesrname';
    const icon = 'twitter';

    const { container } = render(<SocialIcon name={name} link={link} icon={icon} />);

    const nameElement = screen.getByText(name);

    expect(container).toMatchSnapshot();

    expect(nameElement).toBeInTheDocument();
  });
});
