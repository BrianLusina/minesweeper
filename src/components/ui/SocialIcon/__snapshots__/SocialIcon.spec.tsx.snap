// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon should render 1`] = `
<div>
  <a
    class="icon brands fa-twitter"
    href="https://twitter.com/usesrname"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="label"
    >
      Twitter
    </span>
  </a>
</div>
`;
